x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/advanced_step.tsx

```diff
@@ -19,14 +19,19 @@ export const AdvancedStep: FC<CreateAnalyticsStepProps> = ({
   setCurrentStep,
   stepActivated,
 }) => {
+  const showForm = step === ANALYTICS_STEPS.ADVANCED;
+  const showDetails = step !== ANALYTICS_STEPS.ADVANCED && stepActivated === true;
+
+  const dataTestSubj = `mlAnalyticsCreateJobWizardAdvancedStep${showForm ? ' active' : ''}${
+    showDetails ? ' summary' : ''
+  }`;
```

Comment:
Maybe this is worth moving to a helper function, seeing a similar pattern is used across 3 components.